builder/js/ncf.js

```diff
@@ -89,6 +89,9 @@ app.directive('constraint', function($http, $q, $timeout) {
     },
     require: 'ngModel',
     link: function(scope, elm, attrs, ctrl) {
+    if(scope.parameter.constraints.allow_empty_string){
```

Comment:
You need to check if the value is undefined before unless value will always be empty